lib/versioning/semver-coerced/index.ts

```diff
@@ -42,30 +50,41 @@ function getSatisfyingVersion(
   versions: string[],
   range: string
 ): string | null {
-  const coercedVersions = versions.map((version) => {
-    const coercedVersion = semver.coerce(version);
-    return coercedVersion ? coercedVersion.version : null;
-  });
+  const coercedVersions = versions
+    .map((version) => {
+      const coercedVersion = semver.coerce(version);
+      return coercedVersion ? coercedVersion.version : '';
+    })
+    .filter(Boolean);
+
   return semver.maxSatisfying(coercedVersions, range);
 }
 
 function minSatisfyingVersion(
   versions: string[],
   range: string
 ): string | null {
-  const coercedVersions = versions.map((version) => {
-    const coercedVersion = semver.coerce(version);
-    return coercedVersion ? coercedVersion.version : null;
-  });
+  const coercedVersions = versions
+    .map((version) => {
+      const coercedVersion = semver.coerce(version);
+      return coercedVersion ? coercedVersion.version : '';
+    })
+    .filter(Boolean);
```

Comment:
```suggestion
    .map((version) => semver.coerce(version))
    .filter<SemVer>(Boolean);
```
should do the typecheck 😉

Reply:
It won't work cz the `coercedVersions` array needs to be of type `string[]` and not `SemVer[]`.

Reply:
Can use this instead:-
```js
  type ExcludesFalse = <T>(x: T | false | undefined | null) => x is T;
  const coercedVersions = versions
    .map((version) => semver.coerce(version)?.version)
    .filter(Boolean as any as ExcludesFalse);
```

Reply:
ok, found a better way, see updated suggestion